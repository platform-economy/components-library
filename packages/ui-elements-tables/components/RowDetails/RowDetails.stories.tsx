import * as React from 'react';
import { storiesOf } from '@storybook/react';

import { RowDetails } from './index';

const exampleInfo = `
**description or documentation about RowDetails component**

*supports markdown*
`;

const data = [
  { key: 'key', value: 'value' },
  { key: 'key', value: 'value' },
  { key: 'key', value: 'value' },
  { key: 'key', value: 'value' },
  { key: 'key long', value: 'very very very very veery long value for test purposes' },
  { key: 'key', value: 'value' },
];

storiesOf('ui-elements-tables/RowDetails', module)
  .add('RowDetails', () => (
      <div style={{ width: '100vw' }}>
          <RowDetails data={data}/>
      </div>
    ), ({
      info: exampleInfo,
    }),
  );
