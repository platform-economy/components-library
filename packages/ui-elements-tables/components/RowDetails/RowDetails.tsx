import React, { Component } from 'react';
import styled from 'styled-components';

export type RowDetailsProps = {
  data: DetailItem[];
};

export type DetailItem = {
  key: string;
  value: React.ReactNode;
};

const StyledDetails = styled.div`
  display:flex;
  color: ${props => props.theme.palette.text};
  flex-direction: row;
  justify-content: flex-start;
  flex-wrap: wrap;
  & > div {
    min-width: 24em;
    display: flex;
    flex: 0 1 auto;
    padding: .5em 20px .5em 0;
  }
  & > div > span {
    margin-right: 1em;
    min-width: 10em;
  }
  & > div > span.bold {
    min-width: 0;
    font-weight: bold;
    width: auto;
    margin: 0;
  }
  & > div:last-child {
    margin-right:auto;
  }
`;

export class RowDetails extends Component<RowDetailsProps> {
  renderDetails = () => {
    const { data } = this.props;
    return data.map((item: DetailItem, i) => {
      return (
        <div key={i}>
          <span>{item.key}</span>
          <span className="bold">{item.value}</span>
        </div>
      );
    });
  }
  render() {
    return (
      <StyledDetails>
        {this.renderDetails()}
      </StyledDetails>
    );
  }
}
