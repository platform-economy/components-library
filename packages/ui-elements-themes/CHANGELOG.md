# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
