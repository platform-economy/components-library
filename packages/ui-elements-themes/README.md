# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
