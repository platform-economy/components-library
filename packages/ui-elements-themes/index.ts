export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
