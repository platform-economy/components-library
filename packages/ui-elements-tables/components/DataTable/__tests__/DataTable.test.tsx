import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
/*
A set of utilities for testing Styled Components with Jest.
This package improves the snapshot testing experience and
provides a brand new matcher to make expectations on the style rules.
*/
import 'jest-styled-components';
import { withTestTheme } from '@relayr/ui-elements-themes';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });

// Component
import {
  DataTable,
  ColumnType,
  renderDetailsInternally,
  controlIndeterminate,
  isRowExpanded,
  checkItem,
  uncheckItem,
  toggleExpand,
  toggleExpandAll,
  checkAll,
  uncheckAll,
} from '../index';
import { DetailItem, RowDetails } from '../../RowDetails';

type RowItem = {
  test: string;
  id: string;
  details: DetailItem[],
};

type TestRow = {
  test: string,
  id: string,
  details?: unknown[],
};

const renderDetails = (data:React.ReactNode) => {
  return data;
};

const getRowId = jest.fn();

const detailsData = [
  { key: 'Test', value: 'Test' },
];

const getDetails = (data:RowItem) => data.details;

const columns: ColumnType<TestRow>[] = [
  {
    columnId: 'c',
    type: 'checkbox',
    renderCell: (data, handler) => {
      return (
      <input
        type="checkbox"
        // tslint:disable-next-line:jsx-no-lambda
        onChange={e => handler.setChecked(handler.checked)}
        checked={handler.checked}
      />);
    },
    renderCaption: (headerHandler) => {
      return (
      <input
        type="checkbox"
        className="indeterminate"
        // tslint:disable-next-line:jsx-no-lambda
        onChange={e => headerHandler.setChecked(!headerHandler.checked)
        }
        checked={!!headerHandler.checked}
      />);
    },
  },
  {
    columnId: 'c0',
    type: 'control',
    renderCell: () => 'test',
    caption: 'd',
  },
  {
    columnId: 'c1',
    type: 'checkbox',
    renderCell: () => 'test',
    caption: 'd',
  },
  {
    columnId: 'c2',
    renderCell: () => 'test',
    caption: 'c2',
    short: true,
    align: 'center',
  },
  {
    columnId: 'c3',
    renderCell: () => 'test2',
    caption: 'Name',
  },
  {
    columnId: 'c4',
    type: 'empty',
    renderCell: () => '',
  },
  {
    columnId: 'c5',
    renderCell: () => 'test2',
    renderCaption: undefined,
    caption: 'caption',
  },
  {
    columnId: 'c6',
    renderCell: () => 'test4',
    renderCaption: () => 'test4',
  },
  {
    columnId: 'c7',
    renderCell: () => 'test',
  },
  {
    type: 'control',
    columnId: 'c8',
    short: true,
    renderCell: (data, handler) => (
      <button
      // tslint:disable-next-line:jsx-no-lambda
        onClick={() => {
          handler.toggleExpanded();
        }}
      />
    ),
    renderCaption: headerHandler => (
      <button
      // tslint:disable-next-line:jsx-no-lambda
        onClick={() => {
          headerHandler.toggleExpanded();
        }}
      />
    ),
  },
];

const rows = [
  { test: 'Test', id: 'r0', details: [], c2: 'test', c3: 'test', c4:'test', c5:'test', c6:'test' },
  { test: 'Test', id: 'r1', details: detailsData },
  { test: 'Test', id: 'r2' },
];

describe('ui-elements-tables', () => {
  describe('Components', () => {
    describe('DataTable', () => {

      it('should render with renderDetails', () => {
        const mock = jest.fn();
        const tree = TestRenderer
          .create(withTestTheme(
          <DataTable
            columns={columns}
            rows={rows}
            getRowId={mock}
            renderDetails={renderDetails}
          />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render with getDetails', () => {
        const tree = TestRenderer
          .create(withTestTheme(
          <DataTable
            columns={columns}
            rows={rows}
            getRowId={getRowId}
            getDetails={getDetails}
          />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render with no details', () => {
        const tree = TestRenderer
          .create(withTestTheme(
          <DataTable
            columns={columns}
            rows={rows}
            getRowId={getRowId}
          />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render details internally', () => {
        const data:DetailItem[] = [];
        const renderFunction = renderDetailsInternally();
        expect(renderFunction).toEqual(<div>No Details to show</div>);
        expect(renderDetailsInternally(data)).toEqual(<RowDetails data={data} />);
      });

      it('should handle indeterminate state properly', () => {
        const empty:React.ReactText[] = [];
        const one:React.ReactText[] = ['test'];
        const two:React.ReactText[] = ['test', 'test2'];
        expect(controlIndeterminate(empty, empty)).toBeTruthy();
        expect(controlIndeterminate(one, two)).toBe(null);
        expect(controlIndeterminate(empty, two)).toBeFalsy();
      });

      it('should check and uncheck items properly', () => {
        const one = '1';
        const two = '2';
        const checkedItems = ['2'];
        expect(checkItem(one, checkedItems)).toEqual(['2', '1']);
        expect(checkItem(two, checkedItems)).toEqual(['2']);
        expect(uncheckItem(two, checkedItems)).toEqual([]);
        expect(uncheckItem(one, checkedItems)).toEqual(['2']);
      });

      it('should expand row properly', () => {
        const one = '1';
        const two = '2';
        const checkedItems = ['2'];
        expect(isRowExpanded(one, checkedItems)).toBeFalsy();
        expect(isRowExpanded(one, checkedItems, true)).toBeTruthy();
        expect(isRowExpanded(one, checkedItems, false)).toBeFalsy();
        expect(isRowExpanded(two, checkedItems)).toBeTruthy();
      });

      it('should toggle expand properly', () => {
        const on = jest.fn();
        const off = jest.fn();
        const one = '1';
        const two = '2';
        const checkedItems = ['2'];
        toggleExpand(one, on, checkedItems);
        expect(on).toBeCalledTimes(1);
        expect(on).toHaveBeenCalledWith(['2', '1']);
        toggleExpand(two, off, checkedItems);
        expect(off).toBeCalledTimes(1);
        expect(off).toHaveBeenCalledWith([]);
        toggleExpand(two);
      });

      it('should toggle expand all properly', () => {
        const on = jest.fn();
        const off = jest.fn();
        const ids = ['2', '1'];
        const one = ['1'];
        toggleExpandAll(on, ids, one);
        expect(on).toBeCalledTimes(1);
        expect(on).toBeCalledWith(['1', '2']);
        toggleExpandAll(off, ids, undefined);
        expect(off).toBeCalledTimes(1);
        toggleExpandAll(on, ids, ids);
        expect(on).toBeCalledWith([]);
        toggleExpandAll(undefined, ids, one);
      });

      it('should check and uncheck all properly', () => {
        const on = jest.fn();
        const off = jest.fn();
        const ids = ['1', '2'];
        const one = ['1'];
        checkAll(one, ids, on);
        expect(on).toBeCalledTimes(1);
        expect(on).toBeCalledWith(ids);
        uncheckAll(one, ids, off);
        expect(off).toBeCalledTimes(1);
        expect(off).toBeCalledWith([]);
      });

      it('should handle checks and expand properly', () => {
        const WrappingComponent = (props: { children: React.ReactElement }) => (
          withTestTheme(props.children)
        );
        const expand = jest.fn();
        const check = jest.fn();
        const wrapper = enzyme.mount(
          <DataTable
            columns={columns}
            rows={rows}
            // tslint:disable-next-line:jsx-no-lambda
            getRowId={(row:RowItem) => row.id}
            getDetails={getDetails}
            checkedItems={['r0']}
            expandedItems={['r1']}
            onCheckedItemsChange={check}
            onExpandedItemsChange={expand}
          />
        , { wrappingComponent: WrappingComponent },
        );
        wrapper.find('button').at(1).simulate('click');
        expect(expand).toHaveBeenCalledTimes(1);
        wrapper.find('input[type="checkbox"]').at(1).simulate('change');
        expect(check).toHaveBeenCalledTimes(1);
        wrapper.find('input[type="checkbox"]').at(2).simulate('change');
        expect(check).toHaveBeenCalledTimes(2);
        wrapper.find('button').at(0).simulate('click');
      });

      it('should uncheck all properly', () => {
        const WrappingComponent = (props: { children: React.ReactElement }) => (
          withTestTheme(props.children)
        );
        const check = jest.fn();
        const ids = rows.map(row => row.id);
        const wrapper = enzyme.mount(
          <DataTable
            columns={columns}
            rows={rows}
            checkedItems={ids}
            // tslint:disable-next-line:jsx-no-lambda
            getRowId={(row:RowItem) => row.id}
            getDetails={getDetails}
            onCheckedItemsChange={check}
          />
        , { wrappingComponent: WrappingComponent },
        );
        wrapper.find('input[type="checkbox"]').at(0).simulate('change');
        expect(check).toBeCalledTimes(1);
        expect(check).toBeCalledWith([]);
      });
      it('should check all properly', () => {
        const WrappingComponent = (props: { children: React.ReactElement }) => (
          withTestTheme(props.children)
        );
        const check = jest.fn();
        const ids = ['r1'];
        const wrapper = enzyme.mount(
          <DataTable
            columns={columns}
            rows={rows}
            checkedItems={ids}
            // tslint:disable-next-line:jsx-no-lambda
            getRowId={(row:RowItem) => row.id}
            getDetails={getDetails}
            onCheckedItemsChange={check}
          />
        , { wrappingComponent: WrappingComponent },
        );
        wrapper.find('input[type="checkbox"]').at(0).simulate('change');
        expect(check).toBeCalledTimes(1);
        expect(check).toBeCalledWith(['r1', 'r0', 'r2']);
      });
    });
  });
});
