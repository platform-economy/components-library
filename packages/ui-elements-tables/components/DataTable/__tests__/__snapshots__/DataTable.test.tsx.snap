// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-tables Components DataTable should render with getDetails 1`] = `
.c1 {
  background-color: #foreground;
  font-size: .875em;
}

.c5 {
  background-color: #surface;
  font-size: .75em;
}

.c5 > tr:not(:first-child) {
  border-top: solid 2px #midground;
}

tbody > .c2:hover {
  background-color: inherit;
  cursor: inherit;
}

.c8 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 18px 16px;
  text-align: initial;
}

.c8:first-child {
  padding-left: 30px;
}

.c8:last-child {
  padding-right: 30px;
}

.c9 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 18px 16px;
  text-align: center;
}

.c9:first-child {
  padding-left: 30px;
}

.c9:last-child {
  padding-right: 30px;
}

.c6.expanded {
  background-color: #midground;
}

tbody > .c6.clickable:hover {
  background-color: #midground;
  cursor: pointer;
}

.c4 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 22px 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: left;
}

.c4.c4:first-child {
  padding-left: 30px;
}

.c4.c4:last-child {
  padding-right: 30px;
}

.c4.active {
  color: #active;
}

.c4.short {
  width: 1px;
  white-space: nowrap;
  text-align: center;
}

.c4.clickable:hover {
  color: #active_hovered;
  cursor: pointer;
}

.c0 {
  color: #text;
  border-collapse: collapse;
  width: 100%;
  height: 100%;
  box-shadow: 0 1px 3px 0 #shadow;
}

.c0.condensed .c7,
.c0.condensed .c3 {
  padding-top: 10px;
  padding-bottom: 10px;
}

<table
  cellPadding={0}
  cellSpacing={0}
  className="c0"
>
  <thead
    className="c1"
  >
    <tr
      className="c2"
    >
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        <input
          checked={false}
          className="indeterminate"
          onChange={[Function]}
          type="checkbox"
        />
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        d
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        d
      </th>
      <th
        className="short c3 c4"
        onClick={[Function]}
      >
        c2
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        Name
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
         
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        caption
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        test4
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      />
      <th
        className="short c3 c4"
        onClick={[Function]}
      >
        <button
          onClick={[Function]}
        />
      </th>
    </tr>
  </thead>
  <tbody
    className="c5"
  >
    <tr
      className="c6"
    >
      <td
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c9"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        test4
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        <button
          onClick={[Function]}
        />
      </td>
    </tr>
    <tr
      className="c6"
    >
      <td
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c9"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        test4
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        <button
          onClick={[Function]}
        />
      </td>
    </tr>
    <tr
      className="c6"
    >
      <td
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c9"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        test4
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        <button
          onClick={[Function]}
        />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`ui-elements-tables Components DataTable should render with no details 1`] = `
.c1 {
  background-color: #foreground;
  font-size: .875em;
}

.c5 {
  background-color: #surface;
  font-size: .75em;
}

.c5 > tr:not(:first-child) {
  border-top: solid 2px #midground;
}

tbody > .c2:hover {
  background-color: inherit;
  cursor: inherit;
}

.c7 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 18px 16px;
  text-align: initial;
}

.c7:first-child {
  padding-left: 30px;
}

.c7:last-child {
  padding-right: 30px;
}

.c8 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 18px 16px;
  text-align: center;
}

.c8:first-child {
  padding-left: 30px;
}

.c8:last-child {
  padding-right: 30px;
}

.c4 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 22px 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: left;
}

.c4.c4:first-child {
  padding-left: 30px;
}

.c4.c4:last-child {
  padding-right: 30px;
}

.c4.active {
  color: #active;
}

.c4.short {
  width: 1px;
  white-space: nowrap;
  text-align: center;
}

.c4.clickable:hover {
  color: #active_hovered;
  cursor: pointer;
}

.c0 {
  color: #text;
  border-collapse: collapse;
  width: 100%;
  height: 100%;
  box-shadow: 0 1px 3px 0 #shadow;
}

.c0.condensed .c6,
.c0.condensed .c3 {
  padding-top: 10px;
  padding-bottom: 10px;
}

<table
  cellPadding={0}
  cellSpacing={0}
  className="c0"
>
  <thead
    className="c1"
  >
    <tr
      className="c2"
    >
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        <input
          checked={false}
          className="indeterminate"
          onChange={[Function]}
          type="checkbox"
        />
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        d
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        d
      </th>
      <th
        className="short c3 c4"
        onClick={[Function]}
      >
        c2
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        Name
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
         
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        caption
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        test4
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      />
      <th
        className="short c3 c4"
        onClick={[Function]}
      >
        <button
          onClick={[Function]}
        />
      </th>
    </tr>
  </thead>
  <tbody
    className="c5"
  >
    <tr
      className="c2"
      rows={
        Array [
          Object {
            "c2": "test",
            "c3": "test",
            "c4": "test",
            "c5": "test",
            "c6": "test",
            "details": Array [],
            "id": "r0",
            "test": "Test",
          },
          Object {
            "details": Array [
              Object {
                "key": "Test",
                "value": "Test",
              },
            ],
            "id": "r1",
            "test": "Test",
          },
          Object {
            "id": "r2",
            "test": "Test",
          },
        ]
      }
    >
      <td
        className="c6 c7"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
      <td
        className="c6 c7"
      >
        test
      </td>
      <td
        className="c6 c7"
      >
        test
      </td>
      <td
        className="c6 c8"
      >
        test
      </td>
      <td
        className="c6 c7"
      >
        test2
      </td>
      <td
        className="c6 c7"
      >
        
      </td>
      <td
        className="c6 c7"
      >
        test2
      </td>
      <td
        className="c6 c7"
      >
        test4
      </td>
      <td
        className="c6 c7"
      >
        test
      </td>
      <td
        className="c6 c7"
      >
        <button
          onClick={[Function]}
        />
      </td>
    </tr>
    <tr
      className="c2"
      rows={
        Array [
          Object {
            "c2": "test",
            "c3": "test",
            "c4": "test",
            "c5": "test",
            "c6": "test",
            "details": Array [],
            "id": "r0",
            "test": "Test",
          },
          Object {
            "details": Array [
              Object {
                "key": "Test",
                "value": "Test",
              },
            ],
            "id": "r1",
            "test": "Test",
          },
          Object {
            "id": "r2",
            "test": "Test",
          },
        ]
      }
    >
      <td
        className="c6 c7"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
      <td
        className="c6 c7"
      >
        test
      </td>
      <td
        className="c6 c7"
      >
        test
      </td>
      <td
        className="c6 c8"
      >
        test
      </td>
      <td
        className="c6 c7"
      >
        test2
      </td>
      <td
        className="c6 c7"
      >
        
      </td>
      <td
        className="c6 c7"
      >
        test2
      </td>
      <td
        className="c6 c7"
      >
        test4
      </td>
      <td
        className="c6 c7"
      >
        test
      </td>
      <td
        className="c6 c7"
      >
        <button
          onClick={[Function]}
        />
      </td>
    </tr>
    <tr
      className="c2"
      rows={
        Array [
          Object {
            "c2": "test",
            "c3": "test",
            "c4": "test",
            "c5": "test",
            "c6": "test",
            "details": Array [],
            "id": "r0",
            "test": "Test",
          },
          Object {
            "details": Array [
              Object {
                "key": "Test",
                "value": "Test",
              },
            ],
            "id": "r1",
            "test": "Test",
          },
          Object {
            "id": "r2",
            "test": "Test",
          },
        ]
      }
    >
      <td
        className="c6 c7"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
      <td
        className="c6 c7"
      >
        test
      </td>
      <td
        className="c6 c7"
      >
        test
      </td>
      <td
        className="c6 c8"
      >
        test
      </td>
      <td
        className="c6 c7"
      >
        test2
      </td>
      <td
        className="c6 c7"
      >
        
      </td>
      <td
        className="c6 c7"
      >
        test2
      </td>
      <td
        className="c6 c7"
      >
        test4
      </td>
      <td
        className="c6 c7"
      >
        test
      </td>
      <td
        className="c6 c7"
      >
        <button
          onClick={[Function]}
        />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`ui-elements-tables Components DataTable should render with renderDetails 1`] = `
.c1 {
  background-color: #foreground;
  font-size: .875em;
}

.c5 {
  background-color: #surface;
  font-size: .75em;
}

.c5 > tr:not(:first-child) {
  border-top: solid 2px #midground;
}

tbody > .c2:hover {
  background-color: inherit;
  cursor: inherit;
}

.c8 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 18px 16px;
  text-align: initial;
}

.c8:first-child {
  padding-left: 30px;
}

.c8:last-child {
  padding-right: 30px;
}

.c9 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 18px 16px;
  text-align: center;
}

.c9:first-child {
  padding-left: 30px;
}

.c9:last-child {
  padding-right: 30px;
}

.c6.expanded {
  background-color: #midground;
}

tbody > .c6.clickable:hover {
  background-color: #midground;
  cursor: pointer;
}

.c4 {
  -webkit-transition: padding .5s;
  transition: padding .5s;
  padding: 22px 16px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: left;
}

.c4.c4:first-child {
  padding-left: 30px;
}

.c4.c4:last-child {
  padding-right: 30px;
}

.c4.active {
  color: #active;
}

.c4.short {
  width: 1px;
  white-space: nowrap;
  text-align: center;
}

.c4.clickable:hover {
  color: #active_hovered;
  cursor: pointer;
}

.c0 {
  color: #text;
  border-collapse: collapse;
  width: 100%;
  height: 100%;
  box-shadow: 0 1px 3px 0 #shadow;
}

.c0.condensed .c7,
.c0.condensed .c3 {
  padding-top: 10px;
  padding-bottom: 10px;
}

<table
  cellPadding={0}
  cellSpacing={0}
  className="c0"
>
  <thead
    className="c1"
  >
    <tr
      className="c2"
    >
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        <input
          checked={false}
          className="indeterminate"
          onChange={[Function]}
          type="checkbox"
        />
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        d
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        d
      </th>
      <th
        className="short c3 c4"
        onClick={[Function]}
      >
        c2
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        Name
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
         
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        caption
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      >
        test4
      </th>
      <th
        className="c3 c4"
        onClick={[Function]}
      />
      <th
        className="short c3 c4"
        onClick={[Function]}
      >
        <button
          onClick={[Function]}
        />
      </th>
    </tr>
  </thead>
  <tbody
    className="c5"
  >
    <tr
      className="c6"
    >
      <td
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c9"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        test4
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        <button
          onClick={[Function]}
        />
      </td>
    </tr>
    <tr
      className="c6"
    >
      <td
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c9"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        test4
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        <button
          onClick={[Function]}
        />
      </td>
    </tr>
    <tr
      className="c6"
    >
      <td
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c9"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        
      </td>
      <td
        className="c7 c8"
      >
        test2
      </td>
      <td
        className="c7 c8"
      >
        test4
      </td>
      <td
        className="c7 c8"
      >
        test
      </td>
      <td
        className="c7 c8"
      >
        <button
          onClick={[Function]}
        />
      </td>
    </tr>
  </tbody>
</table>
`;
